### PR TITLE
MDB multisrc theme

### PR DESCRIPTION
**Pull request by stevenyomi**
Rewrite and merge ManhuaDB + Maofly into multisrc theme as promised in https://github.com/tachiyomiorg/tachiyomi-extensions/pull/11930#issuecomment-1133655606.
- Better parsing
- Category filters
- Change icon for ManhuaDB
- Chapter upload date for Maofly
- Replace Duktaping [JS `LZString`](https://github.com/pieroxy/lz-string) with [native `LZString`](https://github.com/rufushuang/lz-string4java) for Maofly
- Add rate limit (2 per second)

Checklist:

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by stevenyomi** [comment]
Need to rebase and rename the gradle file after #12244 is merged.

File: multisrc/overrides/mdb/maofly/additional.gradle [hunk 1]
```diff
@@ -0,0 +1,3 @@
+dependencies {
+    implementation 'com.github.softwarevidal:lz-string4java:lz-string4java-1.0.0'
```
Comment (arkon) [inline multisrc/overrides/mdb/maofly/additional.gradle:2]: For reference: https://github.com/softwarevidal/lz-string4java/tree/a659f1f2a5aa794b964fc171c6ee9df53b7083db

A fork of a fork of a fork...

Comment (stevenyomi) [reply multisrc/overrides/mdb/maofly/additional.gradle:2]: True. The original repo does not have a project file to build upon, and I managed to find a fork that has it. I'm not experienced in maven/gradle 😂

Comment (stevenyomi) [reply multisrc/overrides/mdb/maofly/additional.gradle:2]: Oh, the parent fork did publish to maven: https://mvnrepository.com/artifact/com.github.wajda/lzstring4java/0.1 - I got trapped in jitpack...

**Pull request merged**